jsapp/js/components/processing/routes.utils.ts

```diff
@@ -0,0 +1,195 @@
+// This is a collection of various utility functions related to processing
+// routes and navigation.
+
+import {generatePath} from 'react-router-dom';
+import {router} from 'js/router/legacy';
+import {ROUTES, PROCESSING_ROUTES} from 'js/router/routerConstants';
+import {getCurrentPath} from 'js/router/routerUtils';
+
+/**
+ * This is a list of available tabs in Single Processing View. Each tab uses
+ * the same string as the one being used in the matching route.
+ */
+export enum ProcessingTab {
+  Transcript = 'transcript',
+  Translations = 'translations',
+  Analysis = 'analysis',
+}
+
+const TabToRouteMap: Map<ProcessingTab, string> = new Map([
+  [ProcessingTab.Transcript, PROCESSING_ROUTES.TRANSCRIPT],
+  [ProcessingTab.Translations, PROCESSING_ROUTES.TRANSLATIONS],
+  [ProcessingTab.Analysis, PROCESSING_ROUTES.ANALYSIS],
+]);
+
+interface ProcessingRouteParts {
+  assetUid?: string;
+  qpath?: string;
+  submissionEditId?: string;
+  tab?: ProcessingTab;
+}
+
+/**
+ * For given processing path, returns all of it's params and parts. If path is
+ * not provided, function is working on current path.
+ */
+export function getProcessingRouteParts(path?: string): ProcessingRouteParts {
+  // Step 1. Fallback to current path
+  let targetPath = path;
+  if (!targetPath) {
+    targetPath = getCurrentPath();
+  }
+
+  // Step 2. We get the parts of checked path, and we drop any existing query
+  // params to not pollute the outcome
+  const pathArray = targetPath.split('?')[0].split('/');
+
+  // Step 3. Get all indexes of all the parts from the route definition.
+  // These checks are a bit annoyting to have, but we want to avoid false
+  // positives, and path splitting is very vague. The idea is to make sure
+  // that the provided path has all necessary static parts of the processing
+  // route.
+  const defRouteArray = PROCESSING_ROUTES.ANALYSIS.split('/');
+  const formsPartIndex = defRouteArray.indexOf('forms');
+  const uidPartIndex = defRouteArray.indexOf(':uid');
+  const dataPartIndex = defRouteArray.indexOf('data');
+  const processingPartIndex = defRouteArray.indexOf('processing');
+  const qpathPartIndex = defRouteArray.indexOf(':qpath');
+  const submissionPartIndex = defRouteArray.indexOf(':submissionEditId');
+  const tabPartIndex = defRouteArray.indexOf('analysis');
+
+  // Step 4. Make sure all the static parts exist in the checked path
+  if (
+    pathArray[formsPartIndex] !== 'forms' ||
+    pathArray[dataPartIndex] !== 'data' ||
+    pathArray[processingPartIndex] !== 'processing'
+  ) {
+    // Not processing path, so we return "empty" results
+    return {
+      assetUid: undefined,
+      qpath: undefined,
+      submissionEditId: undefined,
+      tab: undefined,
```

Comment:
@LMNTL This is good and makes sense, but it causes some `|| ''` in `singleProcessingStore`…